lib/configure-map.js

```diff
@@ -135,10 +135,24 @@ module.exports = function (original, length, options) {
 			return conf.delete(arg);
 		};
 	}
+	extGet = function () {
+		var id, args = arguments;
+		if (resolve) args = resolve(args);
+		id = get(args);
+		return cache[id];
+	};
+	extHas = function () {
+		var id, args = arguments;
+		if (resolve) args = resolve(args);
+		id = get(args);
+		return conf.has(id);
+	};
 	defineProperties(memoized, {
 		__memoized__: d(true),
 		delete: d(extDel),
-		clear: d(conf.clear)
+		clear: d(conf.clear),
+		get: d(extGet),
```

Comment:
I did the renaming for both `_has` and `_get`. I did not think of the async mode and this is very important.
As you said I would expect to get the actual value, and not the result of an async function (like a Promise).
I started to write a test for what I would expect, but it is obviously failing because I get a promise back instead of the actual value of it, when resolved.
In the actual state of the library, this seems rather complex to implement now. It will be hopefully addressed on v1 :)

Just to show you what I would expect from it, here is the test case I wrote:

```javascript
"has & get on Promise mode": function (a, d) {
    var fn = function (x) {
        return new Promise(function (res) {
            setTimeout(function () {
                res(x);
            }, 1);
        });
    }, mfn;
    mfn = memoize(fn, { promise: true });
    a(mfn.has('foo'), false);
    mfn('foo').then(function () {
        // At the moment, we get a solved promise instead of the value
        a(mfn.get('foo'), 'foo');
        a(mfn.has('foo'), true);
        d();
    });
    // Promise is not resolved, yet
    // At the moment, we get true, and an unresolved promise
    a(mfn.has('foo'), false);
    a(mfn.get('foo'), undefined);
},
```